identity-account-storage/src/storage/stronghold.rs

```diff
@@ -36,15 +38,23 @@ pub struct Stronghold {
 }
 
 impl Stronghold {
+  /// Constructs a Stronghold storage instance.
+  ///
+  /// Arguments:
+  ///
+  /// * snapshot: path to a local Stronghold file, will be created if it does not exist.
+  /// * password: password for the Stronghold file, optional.
+  /// * dropsave: save all changes when the instance is dropped. Default: true.
   pub async fn new<'a, T, U>(snapshot: &T, password: U, dropsave: Option<bool>) -> Result<Self>
   where
     T: AsRef<Path> + ?Sized,
-    U: Into<Option<&'a str>>,
+    U: Into<Option<String>>,
```

Comment:
I think we could already make the password required for the next release. This is optional here because there used to be `set_password` through which it could be set after instantiation. But from looking at the new stronghold interface, a `Vec<u8>` will be required to instantiate the [`KeyProvider`](https://github.com/iotaledger/stronghold.rs/blob/eb07c4a41f02415aacc38dadc6034182129d622c/client_new/src/security/keyprovider.rs#L28), and I don't think we want to map `None` to an empty `Vec`. It doesn't really make sense to use stronghold without a password, so we might as well enforce it and give direct control to users. For testing, one can always pass an empty or random string.

Whether to take a `String` or `Vec<u8>` I'm not sure about, but I think a string might be more convenient?

Reply:
Ah, thanks for the information. I've changed `password` to be required.

I think `String` may be more convenient for now? Alternatively I would suggest `&[u8]` but we cannot zeroize the parameter ourselves then, and `String` appears slightly better than `Vec<u8>` in my opinion.